File: src/main/java/seremis/geninfusion/inventory/slot/ToggleableSlot.java
package seremis.geninfusion.inventory.slot;

import net.minecraft.inventory.IInventory;
import net.minecraft.item.ItemStack;

public class ToggleableSlot extends GISlot {

    protected boolean enabled = true;

    public ToggleableSlot(IInventory inventory, int id, int x, int y) {
        super(inventory, id, x, y);
    }

    @Override
    public boolean isItemValid(ItemStack is) {
        return enabled && super.isItemValid(is);
    }

    public void enable() {
        enabled = true;
    }

    public void disable() {
        enabled = false;
    }
}

File: src/main/scala/seremis/geninfusion/inventory/slot/ToggleableSlot.scala
package seremis.geninfusion.inventory.slot

import net.minecraft.inventory.IInventory
import net.minecraft.item.ItemStack

class ToggleableSlot(inventory: IInventory, id: Int, x: Int, y: Int) extends GISlot(inventory, id, x, y) {

    var enabled: Boolean = true

    override def isItemValid(is: ItemStack): Boolean = enabled && super.isItemValid(is)

    def enable() {
        enabled = true
    }

    def disable() {
        enabled = false
    }
}
